sql/hive/src/test/scala/org/apache/spark/sql/hive/client/FiltersSuite.scala

```diff
@@ -139,6 +163,11 @@ class FiltersSuite extends SparkFunSuite with Logging with PlanTest {
         InSet(a("doublecol", DoubleType),
           Range(1, 20).map(s => Literal(s.toDouble).eval(EmptyRow)).toSet),
         "")
+
+      checkConverted(
+        InSet(a("datecol", DateType),
+          Range(1, 20).map(d => Literal(d, DateType).eval(EmptyRow)).toSet),
+        "(datecol >= 1970-01-02 and datecol <= 1970-01-20)")
```

Comment:
Oh, is this not `datecol >= 1970-01-20 and ...`?

Reply:
Never mind. I found the reason.

Reply:
```scala
scala> import org.apache.spark.sql.types._
import org.apache.spark.sql.types._

scala> import org.apache.spark.sql.catalyst.expressions._
import org.apache.spark.sql.catalyst.expressions._

scala> Range(1, 20).map(d => Cast(Literal(d, DateType), StringType, Some("America/Los_Angeles"))).map(_.eval(EmptyRow))
res13: scala.collection.immutable.IndexedSeq[Any] = Vector(1970-01-02, 1970-01-03, 1970-01-04, 1970-01-05, 1970-01-06, 1970-01-07, 1970-01-08, 1970-01-09, 1970-01-10, 1970-01-11, 1970-01-12, 1970-01-13, 1970-01-14, 1970-01-15, 1970-01-16, 1970-01-17, 1970-01-18, 1970-01-19, 1970-01-20)
```